helix-core/src/main/java/org/apache/helix/manager/zk/ZKHelixAdmin.java

```diff
@@ -1035,6 +1035,135 @@ public ClusterManagementMode getClusterManagementMode(String clusterName) {
         : new ClusterManagementMode(status.getManagementMode(), status.getManagementModeStatus());
   }
 
+  @Override
+  public void setPartitionsToError(String clusterName, String instanceName, String resourceName,
+      List<String> partitionNames) {
+    logger.info("Set partitions {} for resource {} on instance {} in cluster {} to ERROR state.",
+        partitionNames == null ? "NULL" : HelixUtil.serializeByComma(partitionNames), resourceName,
+        instanceName, clusterName);
+    sendStateTransitionMessage(clusterName, instanceName, resourceName, partitionNames,
+        Boolean.FALSE);
```

Comment:
Let's make it more generic. you can create an enum for actions like "reset, error". This function can be leveraged more in the future.

Reply:
sounds good, updated.